transcendence-app/src/auth/auth-provider/infrastructure/db/auth-provider.entity.ts

```diff
@@ -6,10 +6,20 @@ export enum AuthProviderKeys {
   USER_ID = '"userId"',
 }
 
-export class AuthProviderEntity {
-  constructor(
-    public provider: AuthProviderType,
-    public providerId: string,
-    public userId: string,
-  ) {}
+interface AuthProviderData {
+  provider: AuthProviderType;
+  providerId: string;
+  userId: string;
+}
+
+export class AuthProvider {
+  provider: AuthProviderType;
+  providerId: string;
+  userId: string;
+
+  constructor(authProviderData: AuthProviderData) {
+    this.provider = authProviderData.provider;
+    this.providerId = authProviderData.providerId;
+    this.userId = authProviderData.userId;
+  }
```

Comment:
Maybe we could avoid the `*Data` interfaces doing this:

```suggestion
export class AuthProvider {
  provider: AuthProviderType;
  providerId: string;
  userId: string;

  constructor(authProvider: AuthProvider) {
    this.provider = authProvider.provider;
    this.providerId = authProvider.providerId;
    this.userId = authProvider.userId;
  }
```
typescript will correctly infere all fields

Reply:
Discussed by call: some of these classes (like the ChatroomMemberWithUser) need this data because the fields returned by the db collapse. So, despite not neede here, I'll keep them for consistency.